highlight/HighlightsFromPointsExplainer.md

```diff
@@ -11,61 +11,529 @@ This document is intended as a starting point for engaging the community and sta
+- There's no need to deal with ranges and rectangles anymore because that's handled by the API.
+
+You can refer to a full implementation in the [Appendix](#example-code-using-highlightsfrompoint).
+
 ## Alternative Solutions
 ### Event-based API
-It was initially discussed in the CSSWG whether an event-based API to handle interactions with custom Highlights was the most suitable option (see [this issue](https://github.com/w3c/csswg-drafts/issues/7513) for details). The main idea was dispatching events to the Highlights present under the cursor upon user actions. Different approaches were taken into account:
+A [previous explainer](events-explainer.md) proposed an [event](https://dom.spec.whatwg.org/#event)-based API to handle interactions with custom Highlights.
+This has been discussed in the CSSWG too, see [this issue](https://github.com/w3c/csswg-drafts/issues/7513) for details. The main idea was dispatching events to the Highlights present under the cursor upon user actions. Different approaches were taken into account:
 
 1. A separate event is dispatched for each Highlight under the cursor in descending priority order. The default action is to advance to the next Highlight and `preventDefault()` can be used to stop this. This might result in firing an arbitrary number of events for a single user action.
 2. Dispatch a single event for the top-priority Highlight, with the event path including other Highlights and then bubbling to the DOM tree. However, pointer events can currently only target elements, so this approach might break other specification assumptions.
 3. A combination of approaches 1 and 2, dispatching two events per user action: one for Highlights and one for the DOM tree. The Highlight event propagates through overlapping Highlights, and if `preventDefault()` wasn't called in that sequence, a new event is fired on the DOM tree.
 
-Each of these event-based options carries significant complexity. The `highlightsFromPoint()` API was selected since it is much simpler to implement and still satisfies the use cases adequately.
+If we wanted to apply option 3 above to the example of online collaboration described before, instead of adding the event listener to the `document`, we should add it to the highlights themselves. This could be achieved as follows:
+
+```html
+let highlightsHit = false;
+for (highlight of highlightToUsername.keys()) {
+    highlight.addEventListener('click', (event) => {
+        if (!highlightsHit) {
+            resetActiveHighlights();
+            highlightsHit = true;
+        }
+        createActiveHighlight(event.currentTarget, event.pageX, event.pageY);
+    });
+}
+
+document.addEventListener('mousemove', (event) => {
+    cursorBox.style.display = 'none';
+});
+
+document.addEventListener('click', (event) => {
+    if (!highlightsHit) {
+        // If no Highlights were hit, reset them in case there were active ones from previous clicks.

```

Comment:
This seems backwards to me. Wouldn't we need to reset the old active highlights if there *were* some previously hit? And if none were previously hit there would be none to reset.

Similar question for the use of `resetActiveHighlights` in the `click` event handler above.

Reply:
(explained in https://github.com/MicrosoftEdge/MSEdgeExplainers/pull/983/files#r2013085961)